src/ex4/dist/main.js

```diff
@@ -107,31 +126,53 @@ class Main {
         await this.renderItems();
     }
 
-    async changeUrgency(taskNumber, color) {
+    saveOrder = async () => {
+        if (!confirm('Are you sure you want to save this tasks order?')) {
+            return;
+        }
+        await this.item.saveOrder();
+    }
+
+    async changeUrgency(taskNumber, urgency) {
         const taskName = document.getElementById(`task-name ${taskNumber}`)
         const taskValue = taskName.innerHTML;
-        const index = this.item.tasksList.findIndex(value => { return value.name === taskValue });
-        const taskToChange = this.item.tasksList[index];
-        taskName.classList.replace(`${taskToChange.color}-task`, `${color}-task`);
-        taskToChange.color = color;
-
-        let urgency;
-        if (color === "red") urgency = 1;
-        else if (color === "goldenrod") urgency = 2;
-        else if (color === "green") urgency = 3;
+        
+        const taskToChange = this.findTaskIndexByName(taskValue);
+        taskName.classList.replace(`urgency-${taskToChange.urgency}-task`, `urgency-${urgency}-task`);
+
+        const checkbox = document.getElementById(`checkbox-cont${taskNumber}`)
+        checkbox.classList.replace(`urgency-${taskToChange.urgency}-task`, `urgency-${urgency}-task`);
+
         taskToChange.urgency = urgency;
         await this.item.update(taskToChange);
     }
 
-    async sort(value) {
+    findTaskIndexByName(taskName) {
+        const index = this.item.tasksList.findIndex(value => { return value.taskName === taskName });
+        return this.item.tasksList[index];
+    }
+
+    sort(value) {
         this.item.sort(value);
-        await this.item.reWrite(this.item.tasksList);
-        await this.renderItems();
+        this.renderItems();
     }
 
     async updatePendingTasks(number) {
         document.getElementById("pending-tasks").innerHTML = `You have ${number} pending tasks`
     }
+
+    getColor(urgency) {
```

Comment:
add a constant on the top:
```
const URGENCIES_CONFIG = {
    [URGENCIES.HIGH]: {
        color: "red",
        text: "High",
    },
    [URGENCIES.MEDIUM]: {
        color: "goldenrod",
        text: "Medium",
    },
    [URGENCIES.LOW]: {
        color: "green",
        text: "Low",
    },
    [URGENCIES.NONE]: {
        color: "none",
    }
}
```

then switch to
```suggestion
    getColor(urgency) {
        return URGENCIES_CONFIG[urgency].color
    }
```